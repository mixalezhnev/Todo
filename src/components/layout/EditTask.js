import React from 'react';

import './EditTask.res/style.css';

EditTask.propTypes = {
  onSave: React.PropTypes.func.isRequired,
  onCancel: React.PropTypes.func.isRequired,
  children: React.PropTypes.arrayOf(React.PropTypes.node.isRequired)
}

export default function EditTask({ onSave, onCancel, children }) {
  return (
    <div className="my-editTask-component">
      <div className="buttons-container">
        <button onClick={onSave}>Save changes</button>
        <button onClick={onCancel}>Cancel</button>
      </div>
      {children}
    </div>
  );
}
