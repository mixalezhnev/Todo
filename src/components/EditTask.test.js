import React from 'react';
import { shallow } from 'enzyme';

import EditTask from './EditTask';

const setup = () => {
  const props = {
    onSave: () => {},
    onCancel: () => {}
  };

  const enzymeWrapper = shallow(
    <EditTask {...props}>
      <div className='test-class'/>
    </EditTask>
  );

  return {
    props,
    enzymeWrapper
  }
};

describe('components', () => {
  describe('EditTask', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});