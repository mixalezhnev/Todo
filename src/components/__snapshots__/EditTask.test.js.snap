exports[`components EditTask should render self and subcomponents 1`] = `
<div
  className="my-editTask-component">
  <div
    className="buttons-container">
    <button
      onClick={[Function]}>
      Save changes
    </button>
    <button
      onClick={[Function]}>
      Cancel
    </button>
  </div>
  <div
    className="test-class" />
</div>
`;
